src/main/java/ru/javawebinar/topjava/repository/mock/InMemoryMealRepositoryImpl.java

```
@@ -1,44 +1,81 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
 import ru.javawebinar.topjava.model.Meal;
 import ru.javawebinar.topjava.repository.MealRepository;
+import ru.javawebinar.topjava.util.DateTimeUtil;
 import ru.javawebinar.topjava.util.MealsUtil;
-
+import java.time.LocalDateTime;
+import java.time.Month;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import static ru.javawebinar.topjava.repository.mock.InMemoryUserRepositoryImpl.ADMIN_ID;
+import static ru.javawebinar.topjava.repository.mock.InMemoryUserRepositoryImpl.USER_ID;
 
+@Repository
 public class InMemoryMealRepositoryImpl implements MealRepository {
+    private static final Logger log = LoggerFactory.getLogger(InMemoryMealRepositoryImpl.class);
     private Map<Integer, Meal> repository = new ConcurrentHashMap<>();
     private AtomicInteger counter = new AtomicInteger(0);
 
     {
-        MealsUtil.MEALS.forEach(this::save);
+        MealsUtil.MEALS.forEach(m -> save(m, USER_ID));
+        save(new Meal(LocalDateTime.of(2015, Month.MAY, 31, 20, 0), "Админ ужин", 600), ADMIN_ID);
+        save(new Meal(LocalDateTime.of(2015, Month.JUNE, 1, 14, 0), "Админ обед", 500), ADMIN_ID);
     }
 
     @Override
-    public Meal save(Meal meal) {
+    public Meal save(Meal meal, int userId) {
+        Integer mealId = meal.getId();
+        meal.setUserId(userId);
         if (meal.isNew()) {
-            meal.setId(counter.incrementAndGet());
-        }
-        repository.put(meal.getId(), meal);
-        return meal;
+            mealId = counter.incrementAndGet();
+            meal.setId(mealId);
+            repository.put(meal.getId(), meal);
+            log.info("save {}", meal);
+        } else
+            if (get(mealId, userId) == null) {
+                return null;
```

better to throw exception "Meal user not found"